### DISCUSSION TITLE
Repeated table extraction in Markdown output

### DISCUSSION THREAD

**Discussion by Meaveryway** [pymupdf/pymupdf4llm#168] [Q&A]
Hello there, 

Thanks for the wonderful work! this outperforms even most commercial solutions out there!
I have a question regarding tables extraction: when extracting a PDF page that has a table to markdown, it seems that the table's raw text is first extracted and put in place of the table, then the formatted table at the bottom of the page.

Is this the desired output? Why?

**Comment by PiochU19** [comment]
I noticed the same issue while investigating performance improvements. I forked the repository, profiled the code, and found that by skipping markdown table extraction at [this line](https://github.com/pymupdf/RAG/blob/b257182d0ac1ce7fda837e395c5382edf423e400/pymupdf4llm/pymupdf4llm/helpers/pymupdf_rag.py#L811), I was able to process PDFs approximately five times faster, and those PDFs didn't have much tables inside.

@JorjMcKie

**Comment by jamie-lemon** [comment]
So I think this is a bug with the latest version, to avoid the duplication of table content I had to go down a version with:

`pip install -Iv pymupdf4llm==0.0.16`

**Comment by Meaveryway** [comment]
@PiochU19 @jamie-lemon This turned out to be a bug indeed. I reported it as an issue over #171

**Comment by tahitimoon** [comment]
0.0.17 seems to have not fixed this issue
